app/assets/javascripts/admin/addon/controllers/admin-embedding-crawlers.js

```diff
@@ -0,0 +1,38 @@
+import Controller, { inject as controller } from "@ember/controller";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { i18n } from "discourse-i18n";
+
+export default class AdminEmbeddingCrawlersController extends Controller {
+  @service toasts;
+  @controller adminEmbedding;
```

Comment:
Just wondering why this controller needs to be referenced in all the components, is there a lot of shared data?

Reply:
Before splitting into tabs whole form was on one page. When we load the embedding controller we send request to get all the hosts and settings

<img width="385" alt="Screenshot 2024-12-20 at 9 50 20 AM" src="https://github.com/user-attachments/assets/48d510a3-acc4-4802-98ca-549ff1592b1e" />

For a moment I was thinking about splitting that serializer into 3, but for example, the index requires information from the settings form about the default value:
<img width="1175" alt="Screenshot 2024-12-20 at 9 50 57 AM" src="https://github.com/user-attachments/assets/46963473-cceb-4d64-b43c-35587cd5c229" />

Because the serializer is not too big, I decided to keep it as it is and just use relevant information on each tab.

Does it sound ok to you?

Reply:
OK that's fair, I guess if we were doing it from scratch we could split it up nicer, but I think this is a fine tradeoff considering we are making the UI much better here :+1: